src/network/NetworkObserver.spec.ts

```diff
@@ -423,7 +423,78 @@ describe('NetworkObserver', () => {
       verify(networkMock.getResponseBody(deepEqual({ requestId: '1' }))).once();
     });
 
-    it('should not load a response body if content capturing is disabled', async () => {
+    it('should skip a response body when mime is not allowed', async () => {
+      // arrange
+      when(optionsSpy.content).thenReturn(true);
+      when(optionsSpy.includeMimes).thenReturn(['application/json']);
+      when(
+        networkMock.getRequestPostData(deepEqual({ requestId: '1' }))
+      ).thenResolve({
+        postData: ''
+      });
+      when(
+        networkMock.getResponseBody(deepEqual({ requestId: '1' }))
+      ).thenResolve({ body: '<html></html>', base64Encoded: false });
+      when(connectionMock.subscribe(anyFunction())).thenCall(async cb => {
+        await cb(requestWillBeSentEvent);
+        await cb(responseReceivedEvent);
+        await cb(loadingFinishedEvent);
+      });
+      // act
+      await sut.subscribe(callback);
+      // assert
+      verify(
+        networkMock.getResponseBody(deepEqual({ requestId: '1' }))
+      ).never();
+    });
+
+    it('should load a response body when mime is in the list of allowed', async () => {
+      // arrange
+      when(optionsSpy.content).thenReturn(true);
+      when(optionsSpy.includeMimes).thenReturn(['text/html']);
+      when(
+        networkMock.getRequestPostData(deepEqual({ requestId: '1' }))
+      ).thenResolve({
+        postData: ''
+      });
+      when(
+        networkMock.getResponseBody(deepEqual({ requestId: '1' }))
+      ).thenResolve({ body: '<html></html>', base64Encoded: false });
+      when(connectionMock.subscribe(anyFunction())).thenCall(async cb => {
+        await cb(requestWillBeSentEvent);
+        await cb(responseReceivedEvent);
+        await cb(loadingFinishedEvent);
+      });
+      // act
+      await sut.subscribe(callback);
+      // assert
+      verify(networkMock.getResponseBody(deepEqual({ requestId: '1' }))).once();
+    });
+
+    it('should skip a response body when mime is not defined', async () => {
```

Comment:
```suggestion
    it('should load a response body when mime is not defined', async () => {
```

Reply:
The primary cause of the 'mime not defined' is DNS/TCP connectivity issues, which prevent a browser to receive a response. The current statement accurately reflects this.  Please refer to the assert stage:
```
// assert
verify(
  networkMock.getResponseBody(deepEqual({ requestId: '1' }))
).never();
```